app/api/groups/change_password.go

```diff
@@ -0,0 +1,59 @@
+package groups
+
+import (
+	"crypto/rand"
+	"github.com/go-chi/render"
+	"math/big"
+	"net/http"
+	"strings"
+
+	"github.com/France-ioi/AlgoreaBackend/app/service"
+)
+
+func (srv *Service) changePassword(w http.ResponseWriter, r *http.Request) service.APIError {
+	var err error
+	user := srv.GetUser(r)
+
+	groupID, err := service.ResolveURLQueryPathInt64Field(r, "group_id")
+	if err != nil {
+		return service.ErrInvalidRequest(err)
+	}
+
+	if apiError := srv.checkThatUserOwnsTheGroup(user, groupID); apiError != service.NoError {
+		return apiError
+	}
+
+retry:
+	newPassword, err := GenerateGroupPassword()
+	service.MustNotBeError(err)
+
+	// `CREATE UNIQUE INDEX sPassword ON groups(sPassword)` must be done
+	err = srv.Store.Groups().Where("ID = ?", groupID).Updates(map[string]interface{}{"sPassword": newPassword}).Error()
+	if err != nil && strings.Contains(err.Error(), "Duplicate entry") {
+		goto retry
```

Comment:
I would handle the case of broken `GenerateGroupPassword` which would always generate the same value and thus "loop" forever (which would be disastrous: kill the DB and block processes)

Reply:
fixed